### ISSUE TITLE
Macro Templating does not recognize macro in marco file.

### ISSUE THREAD

**Issue by ericxiao251** [sqlfluff/sqlfluff#335]
We are working with DBT and have some wrapper macros around the core DBT marcos, i.e. `config`.

Example:
`models/examples/my_first_dbt_model/my_first_dbt_model.sql`
```sql
{{ shopify_config('table') }}
with source_data as (
    select “hello_world” as hello_world
)
select *
from source_data
```

`marcos/shopify_config.sql`
```sql
{% macro shopify_config(type) %}
    {{ config(materialized='view') }}
{% endmacro %}
```

`.sqlfluff`:
```
[sqlfluff:templater:jinja]
load_macros_from_path=macros
```

---

Running `sqlfluff lint models/examples/my_first_dbt_model/my_first_dbt_model.sql -vv`

we get this error:
```
== [models/examples/my_first_dbt_model/my_first_dbt_model.sql] PARSING
== [models/examples/my_first_dbt_model/my_first_dbt_model.sql] FAIL
L:   - | P:   - |  TMP | Unrecoverable failure in Jinja templating: 'config' is
                       | undefined. Have you configured your variables?
                       | https://docs.sqlfluff.com/en/latest/configuration.html
```

**Comment by alanmcruickshank** [comment]
That *should* work so this is definitely a bug. Can you post the content of `my_first_dbt_model.sql`? I'll set this up as a test case.

**Comment by ericxiao251** [comment]
I've updated my issue to include the SQL for `my_first_dbt_model` :).

**Comment by ericxiao251** [comment]
Hey Alan, do you have an idea when you have time to work on this bug?

**Comment by pwildenhain** [comment]
See also #371 and #381, not sure if this is a duplicate of either of those

**Comment by huntdm07** [comment]
I've hit an edge case similar but I think subtly different to what @ericxiao251 is describing. If a user-defined macro calls another user-defined macro, the nested macro doesn't seem to be parsed correctly.

I have this macro `demo_data_clause` being called from a model:
```sql
{% macro demo_data_clause(test_column) %}
(
  condition_a
  {% if test_column is defined %} 
  or {{ test_static_clause(test_column, demo_only='true') }
  {% endif %}
)
{%- endmacro -%}
```

... and running sqlfluff with the `-vvv` flag leads to this error:
```zsh
== [models/marts/core/demo/book_usage_by_page_demo.sql] PARSING
TEMPLATING RAW [jinja] (models/marts/core/demo/book_usage_by_page_demo.sql)
PARSING (models/marts/core/demo/book_usage_by_page_demo.sql)
== [models/marts/core/demo/book_usage_by_page_demo.sql] FAIL
L:   - | P:   - |  TMP | Unrecoverable failure in Jinja templating:
                       | 'test_isbns_clause' is undefined. Have you configured
                       | your variables?
                       | https://docs.sqlfluff.com/en/latest/configuration.html
```

It seems to be trying to parse this as variable although it's another macro, which is understandable since it looks superficially like a variable with the `{{ }}` syntax. The ideal solution would be to recognise this and recursively resolve any nested macro invocations from the known definitions until the output is flattened down to one file - I'm not sure if this is a simple or complex extension of the existing functionality!

**Comment by NiallRees** [comment]
I believe this to be resolved by #508!
